soc/nxp/imxrt/Kconfig.defconfig

```diff
@@ -14,7 +14,7 @@ config ROM_START_OFFSET
 	default 0x2000 if BOOT_FLEXSPI_NOR || BOOT_SEMC_NOR
 
 config ADC_MCUX_12B1MSPS_SAR
-	default y if HAS_MCUX_12B1MSPS_SAR
+	depends on HAS_MCUX_12B1MSPS_SAR
 	depends on ADC
```

Comment:
actually, this is legacy.  I think the proper fix is to just delete these 3 lines. The drivers/adc/Kconfig.mcux is already enough.  And probably delete the HAS_MCUX_12BSPS_SAR kconfig entirely, should just be driven by DT definitions, not Kconfig